sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -1321,5 +1322,56 @@ async def read_feed_ranges(
 
         feed_ranges = [FeedRangeInternalEpk(Range.PartitionKeyRangeToRange(partitionKeyRange)).to_dict()
                 for partitionKeyRange in partition_key_ranges]
-
         return (feed_range for feed_range in feed_ranges)
+
+    async def get_latest_session_token(self,
+                                       feed_ranges_to_session_tokens: List[Tuple[Dict[str, Any], str]],
+                                       target_feed_range: Dict[str, Any]
+                                       ) -> str:
+        """ **provisional** Gets the the most up to date session token from the list of session token and feed
+        range tuples for a specific target feed range. The feed range can be obtained from a partition key
+        or by reading the container feed ranges. This should only be used if maintaining own session token or else
+        the CosmosClient instance will keep track of session token. Session tokens and feed ranges are
+        scoped to a container. Only input session tokens and feed ranges obtained from the same container.
+        :param feed_ranges_to_session_tokens: List of feed range and session token tuples.
+        :type feed_ranges_to_session_tokens: List[Tuple[Dict[str, Any], str]]
+        :param target_feed_range: feed range to get most up to date session token.
+        :type target_feed_range: Dict[str, Any]
+        :returns: a session token
+        :rtype: str
+        """
+        return get_latest_session_token(feed_ranges_to_session_tokens, target_feed_range)
+
+    async def feed_range_from_partition_key(self, partition_key: PartitionKeyType) -> Dict[str, Any]:
+        """ Gets the feed range for a given partition key.
+        :param partition_key: partition key to get feed range.
+        :type partition_key: PartitionKeyType
+        :returns: a feed range
+        :rtype: Dict[str, Any]
+
+        .. note::
+          For the feed range, even through a Dict has been returned, but in the future,
+          the structure may change. Please just treat it as opaque and do not take any dependence on it.
```

Comment:
This is a bit conversational. I would probably stick to something a bit more direct:
`"The structure of the dict representation of a feed range may vary, including which keys are present. It therefore should only be treated as an opaque value."`

Reply:
Changed